ambari-server/src/main/java/org/apache/ambari/server/orm/entities/BlueprintEntity.java

```diff
@@ -63,17 +64,20 @@
    * Unidirectional one-to-one association to {@link StackEntity}
    */
   @OneToOne
-  @JoinColumn(name = "stack_id", unique = false, nullable = false, insertable = true, updatable = false)
+  @JoinColumn(name = "stack_id", unique = false, nullable = true, insertable = true, updatable = false)
```

Comment:
This field is kept only for compatibility purposes. A blueprint will reference mpacks instead of stacks and it can have multiple mpacks.

Reply:
So if a blueprint references a stack but there are multiple mpacks associated with that stack, how is it determined which to use?

Reply:
I think when mpacks are defined the stack property should be ignored. Another option is to validate that one of the mpack declarations point to the stack.

Do you want to remove the reference to stack?

Reply:
I think that leaving the stack without the mpack makes it ambiguous. Do we have problems removing this field? If we're keeping it only for compatibility, then don't we still have to specify it in addition to the mpack?

Reply:
@jonathan-hurley Ambari upgraded from 2.x needs this data at least for blueprint export, possibly other use cases as well.  We may later add upgrade logic to move the data to some other table, but as a first cut I think making it optional is OK.

Reply:
I had thought that there was going to be a conversion to mpacks for HDP and HDF so that after upgrade, there's going to be references to the newly converted mpacks instead of stacks.